### PR TITLE
Update kirkstone to 2024.4.0 release

### PR DESCRIPTION
**Pull request by hikinggrass**

### CODE REVIEW

**Comment by mhei** [comment]
Currently traveling, so the review will take some days.

File: recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://LICENSE.txt;md5=3d149a8cb6c7dab01d788f477294cebc"
+
+SRC_URI = "git://@github.com/SRombauts/SQLiteCpp.git;branch=master;protocol=https"
```
Comment (barsnick) [inline recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb]: Is the `@` in `git://@github.com/` intentional?

Comment (mhei) [reply recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb]: I don't think so. This should be removed.

File: recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+inherit cmake
+
+# tag 3.3.1
+SRCREV = "fe6086c12679d8744c02fa03e83e75feca2413d8" 
+PR = "r0"
```
Comment (mhei) [inline recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb]: This line should be dropped

File: recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://LICENSE.txt;md5=3d149a8cb6c7dab01d788f477294cebc"
+
+SRC_URI = "git://@github.com/SRombauts/SQLiteCpp.git;branch=master;protocol=https"
+
+inherit cmake
+
+# tag 3.3.1
+SRCREV = "fe6086c12679d8744c02fa03e83e75feca2413d8"
```
Comment (mhei) [inline recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb]: Please remove trailing whitespace

File: recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+S = "${WORKDIR}/git"
+
+FILES:${PN} += "${datadir}"
+FILES:${PN} += "${datadir}/SQLiteCPP"
+FILES:${PN} += "${datadir}/SQLiteCPP/*"
+
+DEPENDS += "sqlite3"
+
+EXTRA_OECMAKE += "-DSQLITECPP_INTERNAL_SQLITE=OFF"
```
Comment (mhei) [inline recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb]: Please remove trailing whitespace

File: recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+# tag 3.3.1
+SRCREV = "fe6086c12679d8744c02fa03e83e75feca2413d8" 
+PR = "r0"
+
+S = "${WORKDIR}/git"
+
+FILES:${PN} += "${datadir}"
+FILES:${PN} += "${datadir}/SQLiteCPP"
+FILES:${PN} += "${datadir}/SQLiteCPP/*"
+
```
Comment (mhei) [inline recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb:14]: The first line should be `FILES:${PN}-dev += "${datadir}"` and the two other lines should be dropped. This should be changed in combination with my comment below, that we should use shared libs in the Yocto build.

File: recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb (hunk 4 above)
Comment (mhei) [inline recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb]: As noted above, I recommend to use this as shared library, so I'd change this line to
```
EXTRA_OECMAKE += " \
    -DSQLITECPP_INTERNAL_SQLITE=OFF \
    -DBUILD_SHARED_LIBS=ON \
"
```

File: recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb (hunk 1 above)
Comment (hikinggrass) [reply recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb]: That was a typo, removed

File: recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb (hunk 2 above)
Comment (hikinggrass) [reply recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb]: removed

File: recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb (hunk 5 above)
Comment (hikinggrass) [reply recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb:14]: changed

File: recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb (hunk 4 above)
Comment (hikinggrass) [reply recipes-extended/sqlitecpp/sqlitecpp_3.3.1.bb]: changed

**Review by mhei — APPROVED** [review]
Thanks for your changes, LGTM

**Review by barsnick — APPROVED** [review]

**Pull request merged**